Tribler/Core/CacheDB/SqliteCacheDBHandler.py

```diff
@@ -979,15 +979,20 @@ def searchNames(self, kws, local=True, keys=None, doSort=True):
             result.extend(channel)
 
             if doSort and result[num_seeders_index] <= 0:
-                dont_sort_list.append(result)
-                results.pop(i)
-
+                dont_sort_list.append((index, result))
 
         if doSort:
```

Comment:
This list can only be poppulated if the inner if doSort trigger, so It's better to include it in the branch with the other if.